src/youtube/js/youtube.js

```diff
@@ -0,0 +1,394 @@
+      if(player == null || parentContainer == null || controller == null)
+      {
+        console.warn("Youtube: Failed to create the player");
+        return;
+      }
+      youtubeVideoContainer = parentContainer;
+      $('head').append("<script type = 'text/javascript' src = 'http://www.youtube.com/iframe_api'></script>");
+      window.onYouTubePlayerAPIReady = function() { onYouTubeIframeAPIReady(); }; 
+      element = new OoyalaYoutubeVideoWrapper();
+      if(element == null) return;
+      element.controller = controller;
+      controller.notify(controller.EVENTS.CAN_PLAY);
+
+      youtubeVideoContainer.append(player);
+      return element;
+    };
+
+    /**
+     * Destroys the video technology factory
+     * @public
+     * @method OoyalaVideoFactory#destroy
+     */
+    this.destroy = function() {
+      this.encodings = [];
+      this.create = function() {};
+    };
+
+    this.maxSupportedElements = -1;
+  };
+
+  /* 
+   *This function creates an <iframe> (YouTube player)
+   *after the API code downloads.
+   * @property {object} youtubePlayer The youtube player is creted and assigned to youtubePlayer.
+   */
+  function onYouTubeIframeAPIReady() {
+    youtubePlayer = new YT.Player('player', {
+      videoId: youtubeID,
+      height: "100%",
+      width: "100%",
+      playerVars: { 'autoplay': 0, 'controls': 0 },
+      events: {
+      'onReady': onPlayerReady,
+      'onPlaybackQualityChange': onPlayerPlaybackQualityChange,
+      'onStateChange': onPlayerStateChange,
+      'onError': onPlayerError
+      }
+    });
+    if(!youtubePlayer)
+    {
+      element.controller.notify(element.controller.EVENTS.ERROR, { "errorcode" : -1 });
+    }
+  };
+
+  /*
+   *The Youtube iframe API will call this function when the video player is ready.
+   * @param {object} event The event from the youtube once the player is ready.
+   */
+  function onPlayerReady(event) {
+    playerReady = true;
+    for(var i = 0; i < javascriptCommandQueue.length; i++) 
+    {
+      if(javascriptCommandQueue[i][0] === "play")
+      { 
+        element.play();
+        hasPlayed = true;
+      }
+      else if(javascriptCommandQueue[i][0] === "seek")
+      {  
+        element.seek(javascriptCommandQueue[i][1])
+      }    
+    }    
+  };
+
+  /*
+   *The Youtube iframe API calls this function when the player's quality changes.
+   * @param {object} event The event from the youtube on player quality change.
+   */
+  function onPlayerPlaybackQualityChange(event) {
+    var vtcBitrate = {
+      id: event.data,
+      width: 0,
+      height: 0,
+      bitrate:event.data 
+    }
+    element.controller.notify(element.controller.EVENTS.BITRATE_CHANGED,vtcBitrate);
+  };
+
+  /*
+   *The Youtube iframe API calls this function when the player's state changes.
+   * @param {object} event The event from the youtube on player state change.
+   */
+  function onPlayerStateChange(event) {
```

Comment:
Should we include OO.log messages for each of these events for easier debugging?